src/dispatch/plugins/dispatch_google/groups/plugin.py

```diff
@@ -86,30 +144,40 @@ def add_member(client: Any, group_key: str, email: str, role: str):
             log.debug(f"Error adding group member. GroupKey={group_key} Body={body} ")
 
 
-def remove_member(client: Any, group_key: str, email: str):
+def remove_member(client: Any, group_email: str, member_email: str):
     """Removes member from google group."""
-    return make_call(client.members(), "delete", groupKey=group_key, memberKey=email)
+    membership_name = get_group_membership_name(client, group_email, member_email)
+    return make_call(client.groups().memberships(), "delete", name=membership_name)
 
 
-def list_members(client: Any, group_key: str, **kwargs):
+def list_members(client: Any, group_email: str, **kwargs):
     """Lists all members of google group."""
-    return make_call(client.members(), "list", groupKey=group_key, **kwargs)
+    group_name = get_group_name(client, group_email)
+    return make_call(client.groups().memberships(), "list", parent=group_name)
 
 
-def create_group(client: Any, name: str, email: str, description: str):
+def create_group(client: Any, name: str, group_email: str, description: str):
     """Creates a new google group."""
-    body = {"email": email, "name": name, "adminCreated": False, "description": description}
-    return make_call(client.groups(), "insert", body=body, delay=3)
+    body = {
```

Comment:
If `GOOGLE_CUSTOMER_ID` is only required for google group, we should be louder when it's not available but we are executing this plugin. Perhaps we should do something like:

 `raise DispatchPluginException("Google Groups plugin requires 'GOOGLE_CUSTOMER_ID' but one has not been defined")`
 
 I don't even know what Google would do in this scenario ` "parent": f"customers/None"`.

Reply:
That makes sense. I can add the exception when customer ID is unavailable.

It raises an error about `parent` being invalid or missing, I believe. I can get back to you with the exact message.

Reply:
This is the error message when None is passed for the customer ID missing, which can be confusing for someone who isn't familiar with the Google workspace API. Regardless, I've added the exception proposed above.

```
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://cloudidentity.googleapis.com/v1/groups?key=REDACTED&alt=json&initialGroupConfig=WITH_INITIAL_OWNER returned "Invalid resource.parent". Details: "[{'@type': 'type.googleapis.com/google.rpc.BadRequest', 'fieldViolations': [{'field': 'resource.parent', 'description': 'Invalid resource.parent'}]}]">
```